### PR TITLE
Add support for autoCAST()ing meta_value orderby for date and number in 3.x

### LINKED ISSUE #3041: Support WP_Query like orderby for meta fields
Should be able to do `orderby => 'field_name.meta_value_num'` and have Pods autocast it as a number. Also should work for dates.

### PR DESCRIPTION
**Pull request by Shelob9**
Fixes #3041 in 3.x

Is 3.x sibling of PR #3043

### CODE REVIEW

File: classes/Pods/Data.php [hunk 1]
```diff
@@ -978,7 +978,20 @@ public function build( $params ) {
...
+						$params->orderby[ $i ] = 'CAST(' . str_replace( '.meta_value_num', '.meta_value', $orderby ) . ' AS DECIMAL)';
+					} elseif ( strpos( $orderby, '.meta_value_date' ) ) {
+						$params->orderby[ $i ] = 'CAST(' . str_replace( '.meta_value_date', '.meta_value', $orderby  ) . ' AS DATE)';
+					}
+
+				}
+
+			}
+			
 			$params->orderby = (array) $params->orderby;
```
Comment (sc0ttkclark) [inline classes/Pods/Data.php]: This should go above the orderby handling, as it casts orderby as an array

Comment (Shelob9) [inline classes/Pods/Data.php]: Hmm thats, how I did it here: https://github.com/pods-framework/pods/commit/f6652895a1274c1305a170d67fec6c4f9593d508#diff-a96f9b1c49edf7ebf160a14d0b7b3f2aR901

$params->orderby is already an array by the time it gets there, at least in 2.x

Comment (sc0ttkclark) [inline classes/Pods/Data.php]: Right, it was good in 2.x, but in this case $params can't be assumed it's an array yet.

Comment (Shelob9) [inline classes/Pods/Data.php]: Fixed via 5ed9757

**Pull request merged**